lib/active_yaml/aliases.rb

```diff
@@ -5,22 +5,31 @@ def self.included(base)
       base.extend(ClassMethods)
     end
 
+    ALIAS_KEY_REGEXP = /^\//i
+
     module ClassMethods
 
       def insert(record)
         super if record.attributes.present?
       end
 
       def raw_data
-        super.reject do |k, v|
-          v.kind_of? Hash and k.match(/^\//i)
+        d = super
+        if d.is_a?(Array)
+          d.reject do |h|
+            h.keys.any? { |k| k.match(ALIAS_KEY_REGEXP) }
+          end
+        else
+          d.reject do |k, v|
+            v.kind_of? Hash and k.match(ALIAS_KEY_REGEXP)
+          end
         end
       end
 
     end
 
     def initialize(attributes={})
-      super unless attributes.keys.index { |k| k.to_s.match(/^\//i) }
+      super unless attributes.keys.index { |k| k.to_s.match(ALIAS_KEY_REGEXP) }
```

Comment:
Keep this the way you coded it. (lets minimize change in this PR)

ASIDE: Do wonder if `.any?` makes sense here instead of `.index`